presto-jdbc/src/main/java/io/prestosql/jdbc/PrestoPreparedStatement.java

```diff
@@ -343,6 +350,28 @@ private String toTimeLiteral(Object value)
         throw invalidConversion(value, "time");
     }
 
+    private void setAsTimeWithTimeZone(int parameterIndex, Object value)
+            throws SQLException
+    {
+        requireNonNull(value, "value is null");
+
+        String literal = toTimeWithTimeZoneLiteral(value);
+        setParameter(parameterIndex, formatLiteral("TIME", literal));
+    }
+
+    private String toTimeWithTimeZoneLiteral(Object value)
+            throws SQLException
+    {
+        if (value instanceof OffsetTime) {
+            return OFFSET_TIME_FORMATTER.format((OffsetTime) value);
+        }
+        if (value instanceof String) {
+            // TODO validate proper format
```

Comment:
Can you just use `OFFSET_TIME_FORMATTER.parse()` for that?

Reply:
Conceptually yes, but it does not support picosecond precision.